pandas/io/parsers.py

```diff
@@ -1791,19 +1791,17 @@ def __init__(self, f, **kwds):
         self.comment = kwds['comment']
         self._comment_lines = []
 
-        if isinstance(f, compat.string_types):
-            f = _get_handle(f, 'r', encoding=self.encoding,
-                            compression=self.compression,
-                            memory_map=self.memory_map)
-            self.handles.append(f)
-        elif self.compression:
-            f = _wrap_compressed(f, self.compression, self.encoding)
-            self.handles.append(f)
-        # in Python 3, convert BytesIO or fileobjects passed with an encoding
-        elif compat.PY3 and isinstance(f, compat.BytesIO):
-            from io import TextIOWrapper
+        add_handle = (
```

Comment:
Another way to do this would be to handle this inside `_get_handle` and return from that function `handles` that should be added. IMO that would be more maintainable, as now this boolean expression here depends on the content of the `_get_handle` function, so better to do it there.